### PR TITLE
IDE: add derive to structs generated from JSON if `serde` is in dependencies

### PR DESCRIPTION
**Pull request by Kobzol**
Improvement of https://github.com/intellij-rust/intellij-rust/pull/7816.

(Changelog is omitted if this gets into the same release as #7816).

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/typing/paste/RsConvertJsonToStructCopyPasteProcessor.kt [hunk 1]
```diff
@@ -142,8 +146,16 @@ private fun createAndInsertStruct(
...
-fun generateStruct(factory: RsPsiFactory, struct: Struct, nameMap: Map<Struct, String>): RsStructItem? {
+fun generateStruct(
+    factory: RsPsiFactory,
+    struct: Struct,
+    nameMap: Map<Struct, String>,
+    hasSerdeDependency: Boolean
+): RsStructItem? {
     val structString = buildString {
+        if (hasSerdeDependency) {
+            append("#[derive(serde::Serialize, serde::Deserialize)]\n")
```
Comment (dima74) [inline src/main/kotlin/org/rust/ide/typing/paste/RsConvertJsonToStructCopyPasteProcessor.kt:157]: Let's add imports and use `#[derive(Serialize, Deserialize)]`

Comment (Kobzol) [reply src/main/kotlin/org/rust/ide/typing/paste/RsConvertJsonToStructCopyPasteProcessor.kt]: Changed.

**Review by dima74 — APPROVED** [review]
bors r+

**Comment by Kobzol** [comment]
bors retry

**Comment by Kobzol** [comment]
Hmm, for some reason tests fail with old Rust.

**Comment by dima74** [comment]
> Hmm, for some reason tests fail with old Rust.

I guess this is because of `edition = "2021"`. Please use `MinRustcVersion` or `2018` edition

**Comment by Kobzol** [comment]
bors retry

**Comment by dima74** [comment]
@Kobzol please also consider adding serde [`rename_all`](https://serde.rs/container-attrs.html#rename_all) attribute. It probably should work like this: we detect style of attributes and add corresponding `rename_all` attribute to the struct if needed. Then we add `rename` attribute to fields which naming style doesn't correspond to detected style

**Pull request merged**